backend/src/main/java/ai/verta/modeldb/ServiceSet.java

```diff
@@ -20,12 +20,16 @@
   public ArtifactStoreService artifactStoreService = null;
   public AuthService authService;
   public RoleService roleService;
+  public App app;
 
   public static ServiceSet fromConfig(Config config) throws IOException {
     ServiceSet set = new ServiceSet();
     set.authService = AuthServiceUtils.FromConfig(config);
     set.roleService = RoleServiceUtils.FromConfig(config, set.authService);
 
+    // Initialize App.java singleton instance
+    set.app = App.getInstance();
```

Comment:
I might be missing something, but why do we need to do this dummy initialization?

Reply:
it was already there when we are initializing artifact store https://github.com/VertaAI/modeldb/pull/1856/files#diff-1dbb0555b7d8fbec12866ae57bb60e96c5a457126dab4cb3445c620450b66e9eL41 but when artifact store is disabled the `App.java` main class was not initialized and it will broken while starting spring boot server from main class https://github.com/VertaAI/modeldb/blob/52fc539481559a541289cbe3d247859e27401312/backend/src/main/java/ai/verta/modeldb/App.java#L94. So i have just moved that line to up to initialize Main class.

Reply:
Let's add a comment whenever we have a non-obvious reason to do something (e.g. it will trigger a side effect).